### ISSUE TITLE
History based language detection doesn't appear to use files in workspace

### ISSUE THREAD

**Issue by lramos15** [microsoft/vscode#143744]
Testing #143576

1. Open the VS Code workspace
2. Don't open any files
3. Create an untitled file
4. Paste 
```ts
class Foo {
   private foo: Foo
   constructor() {
      this.foo = ''
   }
}
```
5. :bug: Even though this is a typescript snippet within a workspace with plenty of typescript files the model doesn't pick it up
6. Open a TS file
7. Repaste that snippet, and notice the model then picks it up.

This leads me to believe that the workspace files either aren't used or are not weighted well in the case of no recently opened files.

**Comment by JacksonKearl** [comment]
Yeah, had to remove workspace inspection late last night because we don't have a diagnostic service in web. https://github.com/microsoft/vscode/commit/fc6bcb7429f09950ebc0b3602dfc9271cd9321b6 will try to find a good workaround.

**Comment by roblourens** [comment]
This snippet

```
class Foo {
   constructor() {
   }
}
```

is detected as Java, even though Java doesn't use the constructor keyword, and the vscode workspace only has 1 .java file. But the original snippet is detected as TS. Is that an issue?

**Comment by JacksonKearl** [comment]
The training for TS needs work. The training data only has 3 instances of `constructor` being used like that so it really has no idea that that's TS. Will handle that as a part of the existing issue for not suggesting until enough confidence is obtained, though it'd be good to source more TS files too (only 67 at the moment).
